pkg/miniogw/gateway-storj.go

```diff
@@ -370,10 +375,52 @@ func (s *storjObjects) CopyObject(ctx context.Context, srcBucket, srcObject, des
 	return s.putObject(ctx, destBucket, destObject, r, serMetaInfo)
 }
 
+// SetupCloseHandler creates a 'listener' on a new goroutine which will notify the
+// program if it receives an interrupt from the OS. We then handle this by calling
+// our clean up procedure and exiting the program.
+func SetupCloseHandler(ctx context.Context) {
```

Comment:
this function won't work. contexts aren't mutable. context.WithCancel is making a new context derived from the old one. it isn't modifying the existing context.

Reply:
removed the function